tutorials/rabi-oscillations.md

```diff
@@ -0,0 +1,187 @@
+
+# Hamiltonian
+if use_rwa:
+    H = wc * a.dag() * a + wa * sm.dag() * sm + g * (a.dag() * sm + a * sm.dag())
+else:
+    H = wc * a.dag() * a + wa * sm.dag() * sm + g * (a.dag() + a) * (sm + sm.dag())
+```
+
+### Create a list of collapse operators that describe the dissipation
+
+We create a list of collapse operators `c_ops`, which is later passed on to `qutip.mesolve`. For each of the three processes one collapse operator is defined.
+
+```python
+c_op_list = []
+
+# Photon annihilation
+rate = kappa * (1 + n_th_a)
+c_op_list.append(np.sqrt(rate) * a)
+
+# Photon creation
+rate = kappa * n_th_a
+c_op_list.append(np.sqrt(rate) * a.dag())
+
+# Atom annihilation
+rate = gamma
+c_op_list.append(np.sqrt(rate) * sm)
+```
+
+### Evolve the system
+
+Here we evolve the system with the Lindblad master equation solver `qutip.mesolve`, and we request that the expectation values of the operators $a^\dagger a$ and $\sigma_+\sigma_-$ are returned by the solver by passing the list `[a.dag()*a, sm.dag()*sm]` as the fifth argument to the solver.
+
+```python
+output = mesolve(H, psi0, tlist, c_op_list, [a.dag() * a, sm.dag() * sm])
+```
+
+### Visualize the results
+
+Here we plot the excitation probabilities of the cavity and the atom (these expectation values were calculated by the `mesolve` above). We can clearly see how energy is being coherently transferred back and forth between the cavity and the atom.
+
+```python
+fig, ax = plt.subplots(figsize=(8,5))
+ax.plot(tlist, output.expect[0], label="Cavity")
+ax.plot(tlist, output.expect[1], label="Atom excited state")
+ax.legend()
+ax.set_xlabel('Time')
+ax.set_ylabel('Occupation probability')
+ax.set_title('Vacuum Rabi oscillations at T={}'.format(n_th_a));
+```
+
+### Non-zero temperature
+Above we set $T = 0$ and thereby discarded the photon creation by the environment. We can activate this term by setting the corresponding variable to a positive value and perform the same calculation as above. In comparison to the previous plot, we see that the cavity has more energy than the atom.
+
+```python
+# set temperature
+n_th_a = 2.0
+
+# set collapse operators
+c_op_list = []
+rate = kappa * (1 + n_th_a)
+c_op_list.append(np.sqrt(rate) * a)
+rate = kappa * n_th_a
+c_op_list.append(np.sqrt(rate) * a.dag())
+rate = gamma
+c_op_list.append(np.sqrt(rate) * sm)
+
+# evolve system
+output = mesolve(H, psi0, tlist, c_op_list, [a.dag() * a, sm.dag() * sm])
+
+# plot
+fig, ax = plt.subplots(figsize=(8,5))
+ax.plot(tlist, output.expect[0], label="Cavity")
+ax.plot(tlist, output.expect[1], label="Atom excited state")
+ax.legend()
+ax.set_xlabel('Time')
+ax.set_ylabel('Occupation probability')
+ax.set_title('Vacuum Rabi oscillations at T={}'.format(n_th_a));
+```
+
+### Software version:
+
+```python
+from qutip import about
+about()
+```
+
+### Testing
+
+```python
+assert np.allclose(output.expect[0][0], 0.0), output.expect[0][0]
+assert np.allclose(output.expect[1][0], 1.0), output.expect[1][0]
```

Comment:
This only checks the initial states -- it would be nice to add a slightly more thorough test if we can. Any thought?

Reply:
We could add one simulation without collapse operators and use the analytical result for the Rabi oscillations. Currently, there is no such section in the notebook, so it would be for testing only.